content/blog/2022/05/2022-05-13-mirrors.jenkins.io-HTTPS.adoc

```diff
@@ -0,0 +1,54 @@
+---
+layout: post
+title: "Enforced HTTPS and consolidation for the Jenkins Mirrors"
+tags:
+- infrastructure
+- mirrors
+- jenkins
+authors:
+- dduportal
+:sig: infra
+description: >
+  The 19th of May 2022, the domain `mirrors.jenkins.io` will be forced to HTTPS and consolidated to the same system as `get.jenkins.io`.
+---
+
+The Jenkins project provides a download mirror infrastructure allowing to download Jenkins packages and plugins from a download server close to your location.
+
+== How Does it Work?
+
+When a download request is emitted to either `mirrors.jenkins.io` or `get.jenkins.io`, an HTTP redirect response to a mirror download server is answered.
+
+The Jenkins infrastructure as a database of existing mirror servers provided by volunteers and donators.
+
+This database is used to select the closest mirror to your location: that is the role of the services `mirrors.jenkins.io` and `get.jenkins.io`.
+
+== Consolidation
+
+The system behind `mirrors.jenkins.io` is using a software named link:https://mirrorbrain.org/["mirrorbrain"] and as been originally used by the Jenkins infrastructure to provide download redirections to mirrors.
+
+But a new system was introduced in 2020: `get.jenkins.io` is using link:https://github.com/etix/mirrorbits["mirrorbits"] which is a great fit for a Kubernetes environment (lightweight, horizontally scalable) and provide a set of nice WebUI features to "see" the mirror list and grading.
+
+INFO: Try it by yourself: check the mirrors providing the latest Jenkins LTS distribution: https://get.jenkins.io/war-stable/2.332.3/jenkins.war?mirrorlist[]
+
+It's been 2 years and the new system is clearly the most reliable: that's why we, the Jenkins Infrastructure team, want to retire the "mirrorbrain" system.
+
+In consequence, the 19th of May 2022, the domain name `mirrors.jenkins.io` will be changed to point to the actual "mirrorbits" system already available at `get.jenkins.io`.
+
+This consolidation ensures that all users downloading a Jenkins package or plugin will use the closest download mirror to their location, without having to maintain 2 distinct systems.
+
+== Enforced HTTPS
+
+The main consequence is that the HTTPS protocol will be enforced for the request to the domain `mirrors.jenkins.io`.
+
+The "mirrorbrain" system currently serving requests behind the domain `mirrors.jenkins.io` is not available through HTTPS, which a safety issue for users downloading files.
+
+The 19th of May 2022, that won't be the case anymore: like `get.jenkins.io`, HTTPS will be available and enforced for the requests going through the former domain `mirrors.jenkins.io`.
```

Comment:
```suggestion
The 19th of May 2022, that won't be the case anymore: like `get.jenkins.io`, HTTPS will be available and enforced for the requests going through the domain `mirrors.jenkins.io`.
```
With "former" I read it as it would be decommissioned.